gapis/replay/profile.go

```diff
@@ -0,0 +1,60 @@
+// Copyright (C) 2017 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package replay
+
+import (
+	"context"
+
+	"github.com/google/gapid/core/data/binary"
+	"github.com/google/gapid/core/log"
+	"github.com/google/gapid/gapis/api"
+	"github.com/google/gapid/gapis/api/transform"
+	"github.com/google/gapid/gapis/replay/builder"
+	"github.com/google/gapid/gapis/replay/value"
+)
+
+// ProfilePostBack is a transform that causes a post back at the end of the replay.
+type ProfilePostBack struct {
+	Res []Result
+}
+
+func (t *ProfilePostBack) Transform(ctx context.Context, id api.CmdID, cmd api.Cmd, out transform.Writer) {
+	out.MutateAndWrite(ctx, id, cmd)
+}
+
+func (t *ProfilePostBack) Flush(ctx context.Context, out transform.Writer) {
```

Comment:
It may make sense sometime to unify all of these PostBack finalizing calls, so that we don't have to duplicate this all over the place.